packages/mobile/src/components/navigation-container/NavigationContainer.tsx

```diff
@@ -11,24 +10,29 @@ import { useSelector } from 'react-redux'
 
 import type { RootScreenParamList } from 'app/screens/root-screen/RootScreen'
 
-import { ThemeContext } from '../theme/ThemeContext'
-
 import { navigationThemes } from './navigationThemes'
-const getAccountUser = accountSelectors.getAccountUser
+const { getAccountUser } = accountSelectors
+const { getTheme, getSystemAppearance } = themeSelectors
 
-type Props = {
+type NavigationContainerProps = {
   children: ReactNode
 }
 /**
  * NavigationContainer contains the react-navigation context
  * and configures linking
  */
-const NavigationContainer = ({ children }: Props) => {
-  const { theme, isSystemDarkMode } = useContext(ThemeContext)
+const NavigationContainer = (props: NavigationContainerProps) => {
+  const { children } = props
+  const theme = useSelector(getTheme)
+  const systemAppearance = useSelector(getSystemAppearance)
   const account = useSelector(getAccountUser)
 
   const navigationTheme =
-    theme === 'auto' ? (isSystemDarkMode ? 'dark' : 'default') : theme
+    theme === 'auto'
```

Comment:
Could we encapsulate this logic in a hook since we are using it in multiple places?

Reply:
yeah i like that, this one is slightly diff tho, since it's 'auto'? but yeah i hear you that this type of logic is in a lot of spots....